### ISSUE TITLE
parser requires CRLF

### ISSUE THREAD

**Issue by ctron** [eikendev/sectxt#16]
The parser seems to require CRLF (instead of line ending) when RFC 9116 seems to allow for both: https://www.rfc-editor.org/rfc/rfc9116#section-2.2

**Comment by eikendev** [comment]
Hi @ctron, thanks for opening this issue. Would you share a more specific example of where sectxt fails to comply with the RFC?

Please note that the referred section is very general and only says which line endings are accepted without specifying the circumstances. The grammar in the RFC is more clear: for the unsigned format you can use either line ending interchangeably, while for the signed format you must use CRLF (with an exception for the cleartext itself).

**Comment by eikendev** [comment]
Please also consider that there is a pending errata to RFC 9116 to make the use of line endings more liberal.

**Comment by ctron** [comment]
It fails parsing Red Hat's security.txt, which I added in the PR https://github.com/eikendev/sectxt/pull/17

If fails when encountering the `\n` (LF only). Which makes sense as the parser uses `crlf` and not `line_end`. That's what the PR fixes.

To my understanding of the RFC, LF only should be possible:

> Every line MUST end with either a carriage return and line feed characters (CRLF / %x0D %x0A) or just a line feed character (LF / %x0A).

**Comment by ctron** [comment]
> Please also consider that there is a pending errata to RFC 9116 to make the use of line endings more liberal.

The quoted/linked RFC says: `Published: April 2022 `, to my understanding that's the current version already.

**Comment by eikendev** [comment]
Hi @ctron, as mentioned, the grammar provided by the RFC is more specific than the section 2.2 you quoted. For this reason, it is currently used as the ground truth for what is regarded as a valid security.txt file.

Note that by sticking to the grammar, you are automatically compliant with the requirement outlined in section 2.2, whereas the opposite is not true.

Also note that as mentioned, [the errata](https://www.rfc-editor.org/errata/rfc9116) is still pending ("reported"), so it has not gone through the full chain of steps required yet for the RFC to be updated.

That said, I see how being overly strict with the file format does not serve a purpose. Therefore, I would consider adding a flag (--strict) for now, until the errata is settled. What do you think?

**Comment by ctron** [comment]
> That said, I see how being overly strict with the file format does not serve a purpose. Therefore, I would consider adding a flag (--strict) for now, until the errata is settled. What do you think?

As I am more interested in the library for parsing a `security.txt`, an option there would be sufficient for my use case. Still a `--strict` flag might be helpful for others.

**Comment by ctron** [comment]
I am wondering though if it's worth the effort to support both ways. Given the fact that there already is an errata pending. Would it be about authoring that file, yes. In accepting it's content, maybe no.

Btw, I checked Suse too (https://www.suse.com/.well-known/security.txt) and they also have LF only. Canonical/Ubuntu don't seem to have one.

**Comment by eikendev** [comment]
I have opened #19 to support a switch in sectxtlib. Can you check if this is what you were looking for?

**Comment by ctron** [comment]
I guess the `strict` tag would work. But I am not sure the idea works in general. It looks like that `strict` turns this into an LF-only parser. While the RFC seems to allow either CRLF or LF.

**Comment by eikendev** [comment]
> While the RFC seems to allow either CRLF or LF.

I don't agree this is fully true, and it's exactly where (as mentioned) the grammar in section 4 is more specific than the statement in section 2.2.

Here an example with the armor header:
```
armor-header     =  %s"-----BEGIN PGP SIGNATURE-----" CRLF
...
CRLF             =  CR LF
```

Note that for an unsigned security.txt file, the carriage return *is* fully optional, and sectxt *does* regard it as such:

```
line             =  [ (field / comment) ] eol
eol              =  *WSP [CR] LF
```

While the errata is not moving forward, we have no indication how the authors intended to specify this. I do admit there is a certain ambiguity, but we can resolve this with a strict mode. As you mentioned you were more interested in the library itself, does the referenced PR resolve this issue?

**Comment by ctron** [comment]
Maybe I just don't understand the code. When having "strict mode off", would if allow parsing a CRLF and a LF file?

**Comment by eikendev** [comment]
Yes. Fixed in 6613ef343b5590136dad4d91f2a234c7f4277800. Please feel free to open a new issue if you think there is still a bug.

**Comment by ctron** [comment]
@eikendev I finally found some time to check it, and works great. Thanks!
